habit-backend/users/views.py

```diff
@@ -0,0 +1,89 @@
+import re
+
+from django.contrib.auth import authenticate, login
+from rest_framework import permissions, generics, status
+from rest_framework.response import Response
+from rest_framework_jwt.settings import api_settings
+from .userhelper import ValidateEmail
+
+from users.models import User
+from users.serializers import UserSerializer , LoginSerializer
+
+jwt_payload_handler = api_settings.JWT_PAYLOAD_HANDLER
+jwt_encode_handler = api_settings.JWT_ENCODE_HANDLER
+
+
+class SignUpUserView(generics.GenericAPIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def post(self, request, *args, **kwargs):
+
+        email = request.data.get('email')
+        if email is None:
+            return Response({'email': ['This field is required.']},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        email = email.lower()
+        email_valid_status = ValidateEmail(email)
+        if email_valid_status == "Invalid":
+            return Response({'email': ['This field is invalid.']},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        password = request.data.get('password')
+        if password is None:
+            return Response({'password': ['This field is required.']},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+
+        first_name = request.data.get('first_name')
+        last_name = request.data.get('last_name')
+
+        if first_name is None:
+            return Response({'first_name': ['This field is required.']},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        if last_name is None:
+            return Response({'last_name': ['This field is required.']},
```

Comment:
Let's not make this required?

Reply:
And what if Api didn't get this parameter not just value?

Reply:
Api will always send you the `last_name` key. 
But it's value can be blank

Reply:
Okay. Will remove this validation.